### DISCUSSION TITLE
jsonPath error using $__isoFrom()

### DISCUSSION THREAD

**Discussion by birgirob** [grafana/grafana-json-datasource#120] [Q&A] [answered]
Hello, 
I'm getting this error when trying to use $__isoFrom() variable in the Fields tab

![image](https://user-images.githubusercontent.com/6920927/120182463-f5396c80-c1fd-11eb-93cc-a7b4b3f74803.png)

The response JSON looks like this:
```yaml
{ 
  "issues": [
    "fields": [
      ...
      "created": "2020-11-20T12:37:36.000+0000"
      ...
    ]
  ] 
}
```

Am I doing something wrong?

**Comment by marcusolsson** [comment, marked as answer]
Does it work if you surround the macro with double quotes?

**Reply by birgirob** [reply]
Thanks for the quick response.

If I use quotes the error goes away but it doesn't return anything

![image](https://user-images.githubusercontent.com/6920927/120214739-6a1f9d00-c224-11eb-9c05-e6baf965d928.png)

Same with single quotes.

It should definitely be returning data since hardcoding the date in the body works:
```yaml
{
    "jql": "issueFunction in subtasksOf('project = MIS') and created > '2020-05-31 15:24'"
}
```

Shame ISO8601 dates aren't valid in Jira JQL, then I could use params instead of body

**Reply by marcusolsson** [reply]
The query looks right, though I see now that the response JSON you posted doesn't seem to be valid JSON. 

The following works for me. I tried changing the time interval as well to confirm that the filter works:

Response: 

```json
{
  "issues": [
    {
      "fields": [
        {
          "created": "2020-11-20T12:37:36.000+0000"
        }
      ]
    }
  ]
}
```

Query:

```js
$.issues[*].fields[?(@.created > "$__isoFrom()")].created
```

**Reply by birgirob** [reply]
Seems like I made a typo in the JSON above, but now that I look closer at the response, "fields" is actually an object not an array. It should still work, right?

Response:

```json
{
  "issues": [
    {
      "fields":
        {
          "created": "2020-11-20T12:37:36.000+0000",
          "someOtherField": "value"
        }
    }
  ]
}
```

**Reply by birgirob** [reply]
Think I got it working. Maybe should have read JSONPath Plus docs earlier, it has some nice examples :)

This is how I am selecting the 'created' field, a bit convoluted but does the job (can you see a way to simplify it? :))

```javascript
$.issues[*].*[?(@property === 'created' && (@ > '$__isoFrom()' && @ < '$__isoTo()'))]^.created
```

Thanks for the help!
